lib/scheduler.pm

```diff
@@ -79,22 +80,21 @@ in the same file than the schedule or in a dedicated file only for data.
 sub parse_test_data {
     my ($schedule) = shift;
     $test_data = {};
-    # return if section is not defined
-    return unless exists $schedule->{test_data};
 
-    if (defined(my $import = $schedule->{test_data}->{$include_tag})) {
-        # Allow both lists and scalar value for import "!include" key
-        if (ref $import eq 'ARRAY') {
-            for my $include (@{$import}) {
-                _import_test_data_from_yaml($include);
-            }
-        }
-        else {
-            _import_test_data_from_yaml($import);
-        }
+    # return if test_data section is not defined in schedule file
+    if (exists $schedule->{test_data}) {
+        # import test data using !include from test_data section in schedule file
+        _import_test_data_included($schedule->{test_data});
+
+        # test_data from schedule file has priority over included data
+        $test_data = {%$test_data, %{$schedule->{test_data}}};
+    }
+
+    # import test data directly from data file
+    if (my $yamlfile = get_var('YAML_DATA')) {
```

Comment:
I'm not 100% about naming, as we call it test data everywhere, and variable is `YAML_DATA`, which might be confusing for some. Maybe we should use `YAML_TEST_DATA`? WDYT?

Reply:
reasonable!